src/styleguide/components/s-theme-selector.vue

```diff
@@ -110,7 +110,18 @@
        * @param {Event} event - The related DOM event.
        */
       onChange(event) {
-        this.$store.commit('session/setTheme', event.target.value);
+        const themeId = event.target.value;
+        const urlParam = new URLSearchParams(window.location.search);
+
+        this.$store.commit('session/setTheme', themeId);
+
+        if (urlParam.has('theme')) {
+          urlParam.delete('theme');
+        }
+
+        urlParam.append('theme', themeId);
+
+        window.history.pushState({}, '', `${window.location.pathname}?${urlParam}`);
```

Comment:
todo: you should use the `?...` part for pushState. If you give it a complete url, you might loose other stuff in the url (like #hash). If your string starts with `?` or `#` the browser will automatically merge it with the current url.

Reply:
@patric-eberle Makes totally sense just to update where the queries start.